packages/api/src/routes/user.ts

```diff
@@ -159,19 +164,18 @@ async function revokeUserProviderAccess(
   provider: any
 ): Promise<void> {
   const providerOAuth2 = providerOAuth2OptionsSchema.safeParse(provider);
-  // TODO #249: implement garmin revoke support
-  // const providerOAuth1 = providerOAuth1OptionsSchema.safeParse(
-  //   req.query.provider
-  // );
+  const providerOAuth1 = providerOAuth1OptionsSchema.safeParse(provider);
+
   if (providerOAuth2.success) {
     await Constants.PROVIDER_OAUTH2_MAP[providerOAuth2.data].revokeProviderAccess(connectedUser);
   } else if (provider === PROVIDER_APPLE) {
     const apple = new Apple();
     await apple.revokeProviderAccess(connectedUser);
-    // } else if (providerOAuth1.success) {
-    //   // await Constants.PROVIDER_OAUTH1_MAP[
-    //   //   providerOAuth1.data
-    //   // ].deregister(connectedUser);
+  } else if (providerOAuth1.success) {
+    const token = connectedUser.dataValues.providerMap?.garmin?.token;
+    if (token) {
+      await Constants.PROVIDER_OAUTH1_MAP[providerOAuth1.data].deregister([token]);
+    }
```

Comment:
Defs NABD, but if we could check for OAuth1 before Apple it would make the code slightly more readable - first check for standards, then specifics.